### ISSUE TITLE
Bugs in Visibility & Category Placement of Gradeables based on dates

### ISSUE THREAD

**Issue by bmcutler** [Submitty/Submitty#5317]
**Describe the bug**
On the main page for all gradeables, we place the gradeables in category based on various dates, whether the assignment has a due date, whether there will be peer grading, if its a team, whether the gradeable should be available to students before grades are released, etc.

I have seen a number of bugs with the inconsistent placement of gradeables in categories
(Sorry I don't have notes on *all* of the specifics)

Most recently... a gradeable with peer components was not accessible to the students for peer grading for an assignment with no due date and/or submission files were not available to students before grades were released.

This one scenario was:  
* instructor uploads a bulk scanned pdf with 2 person teams.
* Peers were assigned to grade starting on date X, grades were set to be released on date Y.
* Students could not see the gradeable between dates X&Y, even though they were supposed to do peer grading.  
* They could see the gradeable & access the peer site after date Y (but then peer grading was "overdue", and students could see partially completed peer grading)
-> to fix this, I went in an unchecked the 'not accessible to students before grades are released' (default setting for bulk upload), and I also changed the setting for 'assignment has a due date' (but I don't know if that was relevant)

We should sit down with all of the dates and all of the options, re-clarify the design & access policy, write more tests of all the corner cases, and fix the implementation :)

Tagging issue #5316 as possibly tangentially related.

**Comment by bmcutler** [comment]
Also upping the priority of this issue.  
@josharsh I think this will be testable with an end-to-end test (& travis) after you have added peer gradeables to the sample course.

**Comment by josharsh** [comment]
@bmcutler Yes, I am working on it! :)

**Comment by bmcutler** [comment]
I made a bunch of gradeables named e.g., "lesson 1", "lesson 2", "lesson 4", "lesson 5" (in that order).
Then the next day I made "lesson 3".  

They are displaying 1 2 4 5 3, even though I've edited the dates of 3 to match the other 4.  I don't know what order they are choosing to display in.  If all the dates match, it should fall back on title of gradeable.
I think it's missing that tiebreak ordering and sticking with some internal database order (which is row creation order or something).
